docs/knowledgebase/advanced/try-runtime.md

```diff
@@ -0,0 +1,184 @@
+[`OnRuntimeUpgradeHelpersExt`][oru-helpers-ext-rustdocs] in order to use `try-runtime` for testing storage migrations. These include:
+
+- **`storage_key`**: Generates a storage key unique to this runtime upgrade. This can be used to communicate data from pre-upgrade to post-upgrade state and check them.
+- **`set_temp_storage`**: Writes some temporary data to a specific storage that can be read (potentially in the post-upgrade hook).
+- **`get_temp_storage`** : Gets temporary storage data written by `set_temp_storage`.
+
+Using the [`frame_executive::Executive`][executive-rustdocs] struct, these helper functions in action would look like:
+
+```rust
+pub struct CheckTotalIssuance;
+impl OnRuntimeUpgrade for CheckTotalIssuance {
+    #[cfg(feature = "try-runtime")]
+    fn post_upgrade() {
+        // iterate over all accounts, sum their balance and ensure that sum is correct.
+    }
+}
+
+pub struct EnsureAccountsWontDie;
+impl OnRuntimeUpgrade for EnsureAccountsWontDie {
+    #[cfg(feature = "try-runtime")]
+    fn pre_upgrade() {
+        let account_count = frame_system::Accounts::<Runtime>::iter().count();
+        Self::set_temp_storage(account_count, "account_count");
+    }
+
+    #[cfg(feature = "try-runtime")]
+    fn post_upgrade() {
+        // ensure that this migration doesn't kill any account.
+        let post_migration = frame_system::Accounts::<Runtime>::iter().count();
+        let pre_migration = Self::get_temp_storage::<u32>("account_count");
+        ensure!(post_migration == pre_migration, "error ...");
+    }
+}
+
+pub type CheckerMigrations = (EnsureAccountsWontDie, CheckTotalIssuance);
+pub type Executive = Executive<_, _, _, _, (CheckerMigrations)>;
+```
+
+### CLI interface
+
+To use `try-runtime` from the command line, run your node with the `--features=try-runtime` flag. 
+
+// TODO: basic commands / common combinations
+
+The possible sub-commands include:
+
+- **`on-runtime-upgrade`**: Executes "tryRuntime_on_runtime_upgrade" against the given runtime state.
+- **`offchain-worker`**: Executes "offchainWorkerApi_offchain_worker" against the given runtime state.
+- **`execute-block`**: Executes "core_execute_block" using the given block and the runtime state of the parent block.
+
+For example, running `try-runtime` with the "on-runtime-upgrade" subcommand on a chain
+running locally:
+
+```bash
+cargo run -- --release --features=try-runtime try-runtime on-runtime-upgrade live ws://localhost:9944
+```
+
+#### Other scenarios
+Using it to re-execute code from a `ElectionProviderMultiPhase` off-chain worker:
+
+```bash
+cargo run -- --release \
+--features=try-runtime \
+try-runtime \
+--execution Wasm \
+--wasm-execution Compiled \
+offchain-worker \
+--header-at 0x491d09f313c707b5096650d76600f063b09835fd820e2916d3f8b0f5b45bec30 \
+live \
+-b 0x491d09f313c707b5096650d76600f063b09835fd820e2916d3f8b0f5b45bec30 \
+-m ElectionProviderMultiPhase
+ws//$HOST:9944
+```
+
```

Comment:
Maybe its too trivial for this, but reminder users that after each (sub)command (`try-runtime` , `offchain-worker`, `live` etc) they can pass the `--help` flag to see the options. ATM the full command is a bit convoluted because you pass different flags to different subcommands, so a little tip might be nice